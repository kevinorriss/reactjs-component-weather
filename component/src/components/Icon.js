import React from 'react'
import PropTypes from 'prop-types'
import clearDayIcon from '../icons/clear-day.png'
import clearNightIcon from '../icons/clear-night.png'
import cloudyIcon from '../icons/cloudy.png'
import fogIcon from '../icons/fog.png'
import partlyCloudyDayIcon from '../icons/partly-cloudy-day.png'
import partlyCloudyNightIcon from '../icons/partly-cloudy-night.png'
import rainIcon from '../icons/rain.png'
import sleetIcon from '../icons/sleet.png'
import snowIcon from '../icons/snow.png'
import windIcon from '../icons/wind.png'
import defaultIcon from '../icons/default.png'

class Icon extends React.Component {
    constructor(props) {
        super(props)
    }

    render() {
        let icon
        switch (this.props.name) {
            case 'clear-day':
                icon = clearDayIcon
                break
            case 'clear-night':
                icon = clearNightIcon
                break
            case 'rain':
                icon = rainIcon
                break
            case 'snow':
                icon = snowIcon
                break
            case 'sleet':
                icon = sleetIcon
                break
            case 'wind':
                icon = windIcon
                break
            case 'fog':
                icon = fogIcon
                break
            case 'cloudy':
                icon = cloudyIcon
                break
            case 'partly-cloudy-day':
                icon = partlyCloudyDayIcon
                break
            case 'partly-cloudy-night':
                icon = partlyCloudyNightIcon
                break
            default:
                icon = defaultIcon
                break
        }

        return (
            <img src={icon} className="weather__icon" />
        )
    }
}

Icon.propTypes = {
    name: PropTypes.string.isRequired,
}

export default Icon
