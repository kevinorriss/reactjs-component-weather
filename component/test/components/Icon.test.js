import Icon from '../../src/components/Icon'

it('should render clear-day Icon correctly', () => {
    const component = shallow(
        <Icon name="clear-day" />
    )
    expect(component).toMatchSnapshot()
})

it('should render clear-night Icon correctly', () => {
    const component = shallow(
        <Icon name="clear-night" />
    )
    expect(component).toMatchSnapshot()
})

it('should render rain Icon correctly', () => {
    const component = shallow(
        <Icon name="rain" />
    )
    expect(component).toMatchSnapshot()
})

it('should render snow Icon correctly', () => {
    const component = shallow(
        <Icon name="snow" />
    )
    expect(component).toMatchSnapshot()
})

it('should render sleet Icon correctly', () => {
    const component = shallow(
        <Icon name="sleet" />
    )
    expect(component).toMatchSnapshot()
})

it('should render wind Icon correctly', () => {
    const component = shallow(
        <Icon name="wind" />
    )
    expect(component).toMatchSnapshot()
})

it('should render fog Icon correctly', () => {
    const component = shallow(
        <Icon name="fog" />
    )
    expect(component).toMatchSnapshot()
})

it('should render cloudy Icon correctly', () => {
    const component = shallow(
        <Icon name="cloudy" />
    )
    expect(component).toMatchSnapshot()
})

it('should render partly-cloudy-day Icon correctly', () => {
    const component = shallow(
        <Icon name="partly-cloudy-day" />
    )
    expect(component).toMatchSnapshot()
})

it('should render partly-cloudy-night Icon correctly', () => {
    const component = shallow(
        <Icon name="partly-cloudy-night" />
    )
    expect(component).toMatchSnapshot()
})

it('should render default Icon correctly', () => {
    const component = shallow(
        <Icon name="anyvaluethatdoesntexist" />
    )
    expect(component).toMatchSnapshot()
})