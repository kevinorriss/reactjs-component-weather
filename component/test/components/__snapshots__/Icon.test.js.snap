// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render clear-day Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M48.999 27.982H51v7.403h-2.001z\\"/><path transform=\\"rotate(45.001 37.716 36.775)\\" d=\\"M34.012 35.773h7.403v2h-7.403z\\"/><path transform=\\"rotate(45.001 62.286 36.772)\\" d=\\"M61.285 33.071h2v7.403h-2z\\"/><path d=\\"M63.674 49.059h7.403v2h-7.403zM28.924 49.059h7.402v2h-7.402zM62.294 50.01c0-.027.004-.054.004-.081h-.008c-.078-6.715-5.558-12.155-12.291-12.155s-12.212 5.44-12.29 12.155h-.006c0 .021.003.041.003.062 0 .027-.004.054-.004.081h.008c.078 6.715 5.558 12.155 12.291 12.155s12.212-5.44 12.29-12.155h.006c0-.022-.003-.042-.003-.062zM50.001 60.227c-5.658 0-10.261-4.586-10.295-10.236.044-5.641 4.643-10.217 10.293-10.217 5.658 0 10.261 4.586 10.295 10.236-.044 5.64-4.643 10.217-10.293 10.217zM49 64.615h2.001v7.402H49z\\"/><path transform=\\"rotate(-134.991 62.288 63.227)\\" d=\\"M58.585 62.227h7.403v2h-7.403z\\"/><path transform=\\"rotate(-134.999 37.715 63.228)\\" d=\\"M36.715 59.526h2v7.403h-2z\\"/></svg>"
/>
`;

exports[`should render clear-night Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M56.003 59.967c-2.495-1.868-4.125-4.832-4.146-8.179.026-3.338 1.654-6.294 4.145-8.159.568-.425 1.18-.793 1.829-1.095a12.39 12.39 0 00-1.829-1.297 12.203 12.203 0 00-6.149-1.666c-6.733 0-12.212 5.44-12.29 12.155h-.006c0 .021.003.041.003.062 0 .027-.004.054-.004.081h.008c.078 6.715 5.558 12.155 12.291 12.155 2.242 0 4.338-.613 6.148-1.666.65-.378 1.263-.811 1.83-1.297a10.231 10.231 0 01-1.83-1.094zm-6.149 2.057c-5.658 0-10.261-4.586-10.295-10.236.044-5.641 4.643-10.217 10.293-10.217 1.544 0 3.004.352 4.319.963-2.601 2.229-4.269 5.516-4.312 9.192h-.006c0 .021.003.041.003.062 0 .027-.004.054-.004.081h.008c.043 3.677 1.712 6.964 4.313 9.192a10.205 10.205 0 01-4.319.963z\\"/></svg>"
/>
`;

exports[`should render cloudy Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M75.363 30.034c-1.3 0-2.58.186-3.819.556-1.803-5.486-6.877-9.216-12.763-9.216-2.854 0-5.64.914-7.935 2.592a18.457 18.457 0 00-9.441-2.592c-10.229 0-18.553 8.322-18.553 18.552 0 1.56.215 3.067.579 4.516-6.854.612-12.247 6.38-12.247 13.391 0 7.418 6.034 13.452 13.451 13.452 2.045 0 4.052-.469 5.884-1.365a13.428 13.428 0 0012.577 8.707c4.644 0 8.922-2.384 11.379-6.279 1.373.316 2.754.476 4.119.476 9.357 0 17.1-6.97 18.354-15.987 6.671-.788 11.865-6.469 11.865-13.35.001-7.419-6.033-13.453-13.45-13.453zM58.595 70.475c-1.443 0-2.914-.207-4.369-.614-.529-.15-1.09.09-1.351.573a11.1 11.1 0 01-9.778 5.844 11.079 11.079 0 01-10.75-8.374 1.17 1.17 0 00-1.139-.887c-.22 0-.439.061-.629.185a11.062 11.062 0 01-5.941 1.735c-6.121 0-11.102-4.982-11.102-11.103 0-5.967 4.732-10.833 10.636-11.08.156-.006.309-.023.466-.023.74 0 1.471.076 2.19.222.674.136 1.337.333 1.983.597a1.177 1.177 0 001.589-.831 11.033 11.033 0 0110.82-8.648c2.591 0 5.115.914 7.106 2.575.397.332.964.364 1.397.08a16.123 16.123 0 018.871-2.655c8.935 0 16.203 7.268 16.203 16.202 0 .093-.013.184-.014.276a16.298 16.298 0 01-.218 2.343c-1.257 7.69-7.929 13.583-15.97 13.583zM77.14 54.434c0-.054.008-.106.008-.161 0-10.23-8.323-18.552-18.553-18.552-3.334 0-6.58.894-9.441 2.592a13.462 13.462 0 00-7.934-2.592c-5.886 0-10.961 3.73-12.763 9.216-.85-.254-1.72-.414-2.601-.494a16.124 16.124 0 01-.653-4.517c0-8.934 7.269-16.202 16.203-16.202 3.158 0 6.226.918 8.871 2.655.434.284 1 .252 1.397-.08a11.113 11.113 0 017.105-2.575c5.222 0 9.67 3.557 10.82 8.648a1.175 1.175 0 001.589.831 11.02 11.02 0 014.174-.819c6.121 0 11.103 4.979 11.103 11.104 0 5.513-4.048 10.091-9.325 10.946z\\"/></svg>"
/>
`;

exports[`should render default Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M63.295 43.878h-.125l-.525 2.774-1.274 3.873h3.748l-1.301-3.898z\\"/><path d=\\"M50 20c-16.568 0-30 13.432-30 30s13.432 30 30 30c16.569 0 30-13.432 30-30S66.568 20 50 20zm-7.448 37.273h-2.499l-6.572-9.247-1.375-2.75h-.125l.399 2.75v9.122h-3.248V39.53h2.499l6.622 9.321 1.325 2.674h.125l-.4-2.674v-9.196h3.249v17.618zm4.848 3.373l-2.449-1.149 8.921-20.143 2.449 1.1L47.4 60.646zm19.868-3.498l-1.398-3.873h-5.373l-1.324 3.873h-3.424l6.297-17.62h2.523l6.299 17.62h-3.6z\\"/></svg>"
/>
`;

exports[`should render fog Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M19.833 60.447h61v2h-61zM19.833 64.447h61v2h-61zM19.833 68.447h61v2h-61zM31.747 52.212h.002l.899.007.212-.716c2.586-8.709 10.741-14.792 19.831-14.792 6.377 0 12.304 2.881 16.259 7.906l.411.521.639-.176a6.828 6.828 0 011.808-.251c3.761 0 6.82 3.06 6.82 6.82a6.79 6.79 0 01-1.16 3.805l-.437.65.525.58c.324.356.582.765.768 1.214l1.848-.764a6.175 6.175 0 00-.656-1.197 8.784 8.784 0 001.112-4.288c0-4.864-3.957-8.82-8.82-8.82-.559 0-1.125.057-1.69.169-4.332-5.2-10.646-8.169-17.427-8.169-9.719 0-18.464 6.343-21.524 15.514a12.328 12.328 0 00-10.418 6.717l1.779.914a10.33 10.33 0 019.219-5.644z\\"/></svg>"
/>
`;

exports[`should render partly-cloudy-day Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M38.999 14.919H41v7.403h-2.001z\\"/><path transform=\\"rotate(45.001 27.715 23.712)\\" d=\\"M24.012 22.71h7.403v2h-7.403z\\"/><path transform=\\"rotate(45.001 52.286 23.71)\\" d=\\"M51.285 20.008h2v7.403h-2z\\"/><path d=\\"M18.924 35.996h7.402v2h-7.402z\\"/><path d=\\"M80.625 47.623c0-4.864-3.957-8.82-8.82-8.82a8.71 8.71 0 00-1.69.169c-4.331-5.2-10.646-8.169-17.427-8.169-.683 0-1.359.039-2.031.101-2.122-3.694-6.099-6.192-10.657-6.192-6.733 0-12.212 5.44-12.29 12.155h-.006c0 .021.003.041.003.062 0 .027-.004.054-.004.081h.008a12.27 12.27 0 003.705 8.645c-.082.222-.176.437-.251.662-6.553.307-11.789 5.735-11.789 12.362 0 6.823 5.551 12.375 12.375 12.375a12.4 12.4 0 005.257-1.171c4.243 4.065 9.782 6.296 15.68 6.296a22.647 22.647 0 0021.249-14.738c.205.023.404.035.603.035a6.093 6.093 0 006.086-6.086 6.04 6.04 0 00-1.111-3.478 8.803 8.803 0 001.11-4.289zM29.706 36.928c.044-5.641 4.643-10.217 10.293-10.217a10.29 10.29 0 018.505 4.502c-7.03 1.341-13.105 5.981-16.235 12.432a10.253 10.253 0 01-2.563-6.717zm47.848 15.73a4.064 4.064 0 011.071 2.73 4.09 4.09 0 01-4.086 4.086c-.305 0-.627-.046-1.014-.145l-.903-.231-.293.887a20.654 20.654 0 01-19.642 14.192c-5.602 0-10.849-2.207-14.775-6.214l-.53-.54-.665.363a10.388 10.388 0 01-4.968 1.266c-5.721 0-10.375-4.654-10.375-10.375s4.653-10.375 10.378-10.375h.002l.885.021.217-.73c.06-.201.137-.393.203-.591.218-.66.468-1.303.749-1.93 2.854-6.369 8.814-10.925 15.693-12.015.675-.107 1.36-.171 2.051-.21.377-.021.754-.047 1.135-.047 6.378 0 12.304 2.881 16.259 7.905l.41.521.64-.174c4.48-1.227 8.629 2.218 8.629 6.569a6.787 6.787 0 01-1.16 3.805l-.437.65.526.582z\\"/></svg>"
/>
`;

exports[`should render partly-cloudy-night Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M81.911 33.854c-3.085.443-6.334-.5-8.715-2.852-2.341-2.379-3.28-5.62-2.838-8.7a10.27 10.27 0 01.519-2.068c-.745.058-1.483.185-2.21.376a12.202 12.202 0 00-8.828 9.228 22.632 22.632 0 00-9.22-1.955c-9.719 0-18.464 6.343-21.524 15.513-6.553.307-11.789 5.735-11.789 12.363 0 6.823 5.551 12.375 12.375 12.375a12.4 12.4 0 005.257-1.171c4.243 4.065 9.782 6.296 15.68 6.296a22.647 22.647 0 0021.249-14.738c.205.023.404.035.603.035a6.093 6.093 0 006.086-6.086 6.04 6.04 0 00-1.111-3.478 8.788 8.788 0 001.111-4.29c0-.673-.083-1.327-.227-1.957a12.233 12.233 0 002.103-1.676 12.186 12.186 0 003.17-5.525c.192-.728.319-1.466.377-2.211-.673.248-1.366.419-2.068.521zm-6.426 15.887a4.063 4.063 0 011.071 2.73 4.09 4.09 0 01-4.086 4.086c-.305 0-.627-.046-1.014-.145l-.903-.231-.293.887A20.654 20.654 0 0150.618 71.26c-5.602 0-10.849-2.207-14.775-6.214l-.53-.54-.665.363a10.388 10.388 0 01-4.968 1.266c-5.721 0-10.375-4.654-10.375-10.375s4.653-10.375 10.369-10.375h.001l.899.007.212-.715c2.586-8.709 10.741-14.792 19.831-14.792 3.157 0 6.202.708 8.954 2.031.686.329 1.352.697 1.998 1.102a20.822 20.822 0 015.307 4.772l.411.521.64-.176a6.828 6.828 0 011.808-.251c2.716 0 5.06 1.6 6.155 3.902.286.602.486 1.25.585 1.932.048.323.08.65.08.986 0 1.363-.401 2.68-1.16 3.807l-.437.65.527.58zm3.534-10.085c-.433.432-.898.809-1.383 1.148-1.442-2.91-4.438-4.92-7.899-4.92-.559 0-1.125.057-1.69.169a22.743 22.743 0 00-6.35-5.283 10.228 10.228 0 012.858-5.575 10.236 10.236 0 013.735-2.374c-.264 3.415.882 6.918 3.451 9.549l-.005.004.046.041c.02.02.035.041.055.06l.006-.005c2.63 2.569 6.135 3.714 9.549 3.451a10.217 10.217 0 01-2.373 3.735z\\"/></svg>"
/>
`;

exports[`should render rain Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M49.329 74.305l-.831 1.238c-.592.883-3.547 5.392-3.547 7.457a4.383 4.383 0 004.378 4.377A4.383 4.383 0 0053.706 83c0-2.064-2.954-6.574-3.546-7.457l-.831-1.238zm0 11.072A2.38 2.38 0 0146.951 83c0-.864 1.211-3.151 2.378-5.048 1.167 1.896 2.377 4.183 2.377 5.048a2.38 2.38 0 01-2.377 2.377zM39.329 74.305l-.831 1.238c-.592.883-3.547 5.392-3.547 7.457a4.383 4.383 0 004.378 4.377A4.383 4.383 0 0043.706 83c0-2.064-2.954-6.574-3.546-7.457l-.831-1.238zm0 11.072A2.38 2.38 0 0136.951 83c0-.864 1.211-3.151 2.378-5.048 1.167 1.896 2.377 4.183 2.377 5.048a2.38 2.38 0 01-2.377 2.377zM59.329 74.305l-.831 1.238c-.592.883-3.547 5.392-3.547 7.457a4.383 4.383 0 004.378 4.377A4.383 4.383 0 0063.706 83c0-2.064-2.954-6.574-3.546-7.457l-.831-1.238zm0 11.072A2.38 2.38 0 0156.951 83c0-.864 1.211-3.151 2.378-5.048 1.167 1.896 2.377 4.183 2.377 5.048a2.38 2.38 0 01-2.377 2.377zM77.865 42.813c0-4.864-3.957-8.82-8.82-8.82-.559 0-1.125.057-1.69.169-4.332-5.2-10.646-8.169-17.427-8.169-9.719 0-18.464 6.343-21.524 15.514-6.553.307-11.789 5.735-11.789 12.362 0 6.823 5.551 12.375 12.375 12.375a12.4 12.4 0 005.257-1.171c4.243 4.065 9.782 6.296 15.68 6.296a22.647 22.647 0 0021.249-14.738c.205.023.404.035.603.035a6.093 6.093 0 006.086-6.086 6.037 6.037 0 00-1.111-3.478 8.784 8.784 0 001.111-4.289zm-3.071 5.037a4.063 4.063 0 011.071 2.73 4.09 4.09 0 01-4.086 4.086c-.305 0-.627-.046-1.014-.145l-.903-.231-.293.887a20.654 20.654 0 01-19.642 14.192c-5.602 0-10.849-2.207-14.775-6.214l-.53-.54-.665.363a10.388 10.388 0 01-4.968 1.266c-5.721 0-10.375-4.654-10.375-10.375s4.653-10.375 10.369-10.375h.001l.899.007.212-.715c2.586-8.709 10.741-14.792 19.831-14.792 6.377 0 12.304 2.881 16.259 7.906l.411.521.64-.176a6.828 6.828 0 011.808-.251c3.761 0 6.82 3.06 6.82 6.82a6.79 6.79 0 01-1.16 3.806l-.437.65.527.58z\\"/></svg>"
/>
`;

exports[`should render sleet Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M82.471 33.108c0-4.863-3.957-8.82-8.82-8.82-.559 0-1.125.057-1.69.169-4.332-5.2-10.646-8.169-17.427-8.169a22.684 22.684 0 00-18.732 9.905l.826.564-.826-.564a22.562 22.562 0 00-2.771 5.614 12.36 12.36 0 00-10.344 6.519 12.4 12.4 0 00-1.464 5.838c0 6.824 5.551 12.375 12.375 12.375 1.818 0 3.621-.402 5.258-1.17 4.242 4.064 9.781 6.295 15.68 6.295a22.647 22.647 0 0021.249-14.738c.205.023.404.035.603.035a6.093 6.093 0 006.086-6.085 6.037 6.037 0 00-1.111-3.478 8.818 8.818 0 001.108-4.29zm-3.072 5.038a4.064 4.064 0 011.071 2.73 4.09 4.09 0 01-4.086 4.085 4 4 0 01-1.014-.145l-.903-.23-.293.887a20.653 20.653 0 01-19.642 14.191c-5.602 0-10.849-2.207-14.775-6.213l-.529-.541-.664.363a10.389 10.389 0 01-4.969 1.266c-5.721 0-10.375-4.654-10.375-10.375 0-1.703.424-3.396 1.227-4.893a10.363 10.363 0 019.142-5.482h.002c.051.004.101.008.153.008l.731-.086.281-.786-1.011-.291v-.019l1.011.31a20.6 20.6 0 012.421-5.167h.168v-.267c.037-.055.068-.114.106-.168a20.685 20.685 0 0117.081-9.034c6.377 0 12.304 2.881 16.259 7.906l.411.521.64-.176a6.828 6.828 0 011.808-.251c3.761 0 6.82 3.06 6.82 6.82 0 1.363-.401 2.68-1.16 3.807l-.437.65.526.58zM37.327 69.749c-2.495 0-4.524-2.029-4.524-4.524s2.029-4.524 4.524-4.524 4.524 2.029 4.524 4.524-2.03 4.524-4.524 4.524zm0-7.049c-1.392 0-2.524 1.133-2.524 2.524s1.132 2.524 2.524 2.524 2.524-1.133 2.524-2.524-1.133-2.524-2.524-2.524zM44.376 83.553a4.529 4.529 0 01-4.524-4.524c0-2.494 2.03-4.523 4.524-4.523s4.524 2.029 4.524 4.523a4.53 4.53 0 01-4.524 4.524zm0-7.048c-1.392 0-2.524 1.132-2.524 2.523s1.132 2.524 2.524 2.524c1.392 0 2.524-1.133 2.524-2.524s-1.132-2.523-2.524-2.523zM54.948 73.273a4.529 4.529 0 01-4.524-4.524 4.528 4.528 0 014.524-4.523 4.528 4.528 0 014.523 4.523 4.528 4.528 0 01-4.523 4.524zm0-7.047c-1.392 0-2.524 1.132-2.524 2.523s1.133 2.524 2.524 2.524 2.523-1.133 2.523-2.524-1.131-2.523-2.523-2.523zM72.765 69.749c-2.494 0-4.523-2.029-4.523-4.524s2.029-4.524 4.523-4.524 4.523 2.029 4.523 4.524-2.029 4.524-4.523 4.524zm0-7.049c-1.392 0-2.523 1.133-2.523 2.524s1.132 2.524 2.523 2.524 2.523-1.133 2.523-2.524-1.132-2.524-2.523-2.524zM65.716 83.553a4.528 4.528 0 01-4.523-4.524c0-2.494 2.029-4.523 4.523-4.523s4.523 2.029 4.523 4.523a4.529 4.529 0 01-4.523 4.524zm0-7.048c-1.392 0-2.523 1.132-2.523 2.523s1.132 2.524 2.523 2.524 2.523-1.133 2.523-2.524-1.132-2.523-2.523-2.523z\\"/></svg>"
/>
`;

exports[`should render snow Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M82.471 33.108c0-4.863-3.957-8.82-8.82-8.82-.559 0-1.125.057-1.69.169-4.332-5.2-10.646-8.169-17.427-8.169a22.684 22.684 0 00-18.732 9.905l.826.564-.826-.564a22.562 22.562 0 00-2.771 5.614 12.36 12.36 0 00-10.344 6.519 12.4 12.4 0 00-1.464 5.838c0 6.824 5.551 12.375 12.375 12.375 1.818 0 3.621-.402 5.258-1.17 4.242 4.064 9.781 6.295 15.68 6.295a22.647 22.647 0 0021.249-14.738c.205.023.404.035.603.035a6.093 6.093 0 006.086-6.085 6.037 6.037 0 00-1.111-3.478 8.818 8.818 0 001.108-4.29zm-3.072 5.038a4.064 4.064 0 011.071 2.73 4.09 4.09 0 01-4.086 4.085 4 4 0 01-1.014-.145l-.903-.23-.293.887a20.653 20.653 0 01-19.642 14.191c-5.602 0-10.849-2.207-14.775-6.213l-.529-.541-.664.363a10.389 10.389 0 01-4.969 1.266c-5.721 0-10.375-4.654-10.375-10.375 0-1.703.424-3.396 1.227-4.893a10.363 10.363 0 019.142-5.482h.002c.051.004.101.008.153.008l.731-.086.281-.786-1.011-.291v-.019l1.011.31a20.6 20.6 0 012.421-5.167h.168v-.267c.037-.055.068-.114.106-.168a20.685 20.685 0 0117.081-9.034c6.377 0 12.304 2.881 16.259 7.906l.411.521.64-.176a6.828 6.828 0 011.808-.251c3.761 0 6.82 3.06 6.82 6.82 0 1.363-.401 2.68-1.16 3.807l-.437.65.526.58zM61.778 75.436a6.696 6.696 0 00-.971-3.471l2.084-2.085-2.215-2.215-2.066 2.065a6.69 6.69 0 00-1.999-.838v-2.911h-3.132v2.911a6.664 6.664 0 00-1.999.838l-2.065-2.065L47.2 69.88l2.084 2.084a6.69 6.69 0 00.056 7.037l-2.14 2.141 2.215 2.215 2.16-2.159a6.637 6.637 0 001.904.78v3.063h3.132v-3.063a6.622 6.622 0 001.905-.781l2.159 2.16 2.216-2.215L60.75 79a6.69 6.69 0 001.028-3.564zm-6.733 2.839a2.84 2.84 0 110-5.68 2.84 2.84 0 010 5.68zM43.277 68.19a6.696 6.696 0 00-.971-3.471l2.084-2.085-2.215-2.215-2.066 2.065a6.69 6.69 0 00-1.999-.838v-2.911h-3.132v2.911a6.664 6.664 0 00-1.999.838l-2.065-2.065-2.215 2.215 2.084 2.084a6.69 6.69 0 00.056 7.037l-2.141 2.141 2.215 2.215 2.16-2.159a6.637 6.637 0 001.904.78v3.063h3.132v-3.063a6.622 6.622 0 001.905-.781l2.159 2.16 2.216-2.215-2.142-2.142a6.668 6.668 0 001.03-3.564zm-6.734 2.84a2.84 2.84 0 110-5.68 2.84 2.84 0 010 5.68zM79.833 68.19a6.696 6.696 0 00-.971-3.471l2.084-2.085-2.215-2.215-2.066 2.065a6.69 6.69 0 00-1.999-.838v-2.911h-3.132v2.911a6.664 6.664 0 00-1.999.838L67.47 60.42l-2.215 2.215 2.084 2.084a6.69 6.69 0 00.056 7.037l-2.141 2.141 2.215 2.215 2.16-2.159a6.637 6.637 0 001.904.78v3.063h3.132v-3.063a6.622 6.622 0 001.905-.781l2.159 2.16 2.216-2.215-2.142-2.142a6.674 6.674 0 001.03-3.565zM73.1 71.03a2.84 2.84 0 110-5.68 2.84 2.84 0 010 5.68z\\"/></svg>"
/>
`;

exports[`should render wind Icon correctly 1`] = `
<img
  className="weather__icon"
  src="<svg version=\\"1.1\\" id=\\"Layer_1\\" xmlns=\\"http://www.w3.org/2000/svg\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\" xml:space=\\"preserve\\"><path d=\\"M28.267 46.752h21.771c.136.012.266.041.403.041a4.993 4.993 0 10-4.993-4.993c0 .958.283 1.843.751 2.603H27.093v2.326h1.063c.038.004.071.023.111.023zM57 53.228c-.033-.002-.063-.021-.098-.021H29.001c-.052 0-.097.023-.148.03v-.03h-1.761v2.361h1.761v-.041c.051.007.095.03.148.03h23.583a4.949 4.949 0 00-.771 2.643 4.992 4.992 0 109.984 0c0-2.689-2.133-4.868-4.797-4.972z\\"/><path d=\\"M67.916 41.222a4.992 4.992 0 00-4.992 4.993c0 .974.289 1.875.771 2.643h-35.33c-.035 0-.065.017-.099.02v-.031h-1.174v2.361h1.174v-.02c.034.003.064.02.099.02h39.648c.035 0 .063-.017.098-.02 2.664-.105 4.797-2.283 4.797-4.973a4.992 4.992 0 00-4.992-4.993zM27.093 40.314h12.57v2h-12.57zM27.343 57.2h19.028v2H27.343z\\"/></svg>"
/>
`;
